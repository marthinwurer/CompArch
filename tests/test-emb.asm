; test the emb instruction
	.entry	start
;
	.org	0x100
one:	.word	0x1
five:	.word	0x5
neg1:	.word	0x1ffff
neg5:	.word	0x1fffb
;
	.org	0x150
swap:	.word	0x00ace
;
	.org	0x200
start:
	ldb	one
	emb	swap
	emb	swap
;
	ldb	five
	emb	swap
	emb	swap
;
	ldb	neg1
	emb	swap
	emb	swap
;
	ldb	neg5
	emb	swap
	emb	swap
;
	halt
