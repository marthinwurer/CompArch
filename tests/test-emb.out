CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 200
200:  11100  LDB   100  A[00000]  B[00000]  XR[000]
201:  15150  EMB   150  A[00000]  B[00001]  XR[000]
202:  15150  EMB   150  A[00000]  B[00ace]  XR[000]
203:  11101  LDB   101  A[00000]  B[00001]  XR[000]
204:  15150  EMB   150  A[00000]  B[00005]  XR[000]
205:  15150  EMB   150  A[00000]  B[00ace]  XR[000]
206:  11102  LDB   102  A[00000]  B[00005]  XR[000]
207:  15150  EMB   150  A[00000]  B[1ffff]  XR[000]
208:  15150  EMB   150  A[00000]  B[00ace]  XR[000]
209:  11103  LDB   103  A[00000]  B[1ffff]  XR[000]
20a:  15150  EMB   150  A[00000]  B[1fffb]  XR[000]
20b:  15150  EMB   150  A[00000]  B[00ace]  XR[000]
20c:  00000  HALT       A[00000]  B[1fffb]  XR[000]

Machine Halted - HALT instruction executed

Simulated time 84 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
